### ISSUE TITLE
Method getElementsByIndexLessThan(index) delivers the root element, too

### ISSUE THREAD

**Issue by JurMarky** [jhy/jsoup#778]
Hi,
I think this method _getElementsByIndexLessThan(index)_ delivers more than the siblings only.
It delivers the parent of a siblings and the siblings of the parent. so there's confusion with the index-
evaluation.
f.e: [JSoupSiblingTest.zip](https://github.com/jhy/jsoup/files/553360/JSoupSiblingTest.zip)

```
public static void main(String[] args) {
    Document doc = Jsoup.parse(
            "<html><body><div><p>before1</p><p>before2</p><p>XXX</p><p>after1</p><p>after2</p></div></body></html>",
            "");
    Element body = doc.select("body").first();
    Elements elems = body.select("p:matchesOwn(XXX)");
    Element xElem = elems.first();
    Elements beforeX = xElem.parent().getElementsByIndexLessThan(xElem.elementSiblingIndex());

    int i = 0;
    for(Element p : beforeX) {
        p.remove();
    }

    System.out.println(body);


    Document doc2 = Jsoup.parse(
            "<html><body><div><p>before1</p><p>before2</p><p>XXX</p><p>after1</p><p>after2</p></div></body></html>",
            "");
    body = doc2.select("body").first();
    elems = body.select("p:matchesOwn(XXX)");
    xElem = elems.first();
    Elements afterX = xElem.parent().getElementsByIndexGreaterThan(xElem.elementSiblingIndex());

    for(Element p : afterX) {
        p.remove();
    }

    System.out.println(body);
}
```

**Output:**
`LesserThan: <body></body>`
`GreaterThan: <body><div><p>before1</p><p>before2</p><p>XXX</p></div></body>`

**Expected:**
`LesserThan: <body><div><p>XXX</p><p>after1</p><p>after2</p></div></body>`
`GreaterThan: <body><div><p>before1</p><p>before2</p><p>XXX</p></div></body>`

**Comment by damonwong93** [comment]
Just did some digging, and it seems like the root element is actually being added to the beforeX and afterX array lists when you call the getElementsByIndexLessThan() and getElementsByIndexGreaterThan() methods.  This seems to happen because of the matches() method inside the IndexLessThan class, which is called for each element when traversing with NodeTraversor:
```
public boolean matches(Element root, Element element) {
    return element.elementSiblingIndex() < index;
}
```

When this method is called for the first time, "node" is actually the root node.  And according to the eval criteria within the matches method, since this root node's sibling index is 0, it actually passes the test.  Thus, the entire root node is added to the array list, which in the case of "LesserThan" wipes out the entire body.  

A solution I found to this was to simply add a check to see if the element being evaluated is the root node.  If it is the root node, matches() return false, which then doesn't add the root node to the array list.  

```
public boolean matches(Element root, Element element) {
    return root != element && element.elementSiblingIndex() < index;
}
```

Thus, the output is what is expected:
```
LesserThan: <body><div><p>XXX</p><p>after1</p><p>after2</p></div></body>
GreaterThan: <body><div><p>before1</p><p>before2</p><p>XXX</p></div></body>
```

This is the most low-level solution I can think of, which targets the actual IndexLessThan class, so as to not affect any other traverse scenarios.  Even so, I haven't done extensive testing on this, so I'm not completely sure it doesn't break anything else.  This is, in fact, my first contribution to Jsoup, so please let me know if you find that this solution causes other problems elsewhere.

**Comment by jhy** [comment]
Thanks @damonwong93  - do you want to submit a PR? You could add another testcase to validate it succeeds, and there should be enough existing testcases to see that it doesn't break other cases (or you could add a couple).

**Comment by jhy** [comment]
Merged with #918 

Thanks!
